### PR TITLE
ENH: Ignore PyCharm project files.

### PR DESCRIPTION
**Pull request by jakirkham**
Ignore PyCharm project files in git.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,7 @@
 *$
 *.bak
 *.diff
+.idea
```
Comment (charris) [inline .gitignore]: Need a `*`?

Comment (charris) [inline .gitignore]: The [recommendation](https://github.com/github/gitignore/blob/master/Global/JetBrains.gitignore) seems to be `.idea/`.

Comment (jakirkham) [inline .gitignore]: Thanks for checking. Not in this case though. It's the folder name just as `.git`.

Comment (charris) [inline .gitignore]: And also `*.iml`.

Comment (jakirkham) [inline .gitignore]: Sure, I can add a `/`. PyCharm doesn't seem to generate the other ones, but I can add them. I think there is a Python plugin for IntelliJ.

Comment (jakirkham) [inline .gitignore]: Do we want any of the other ones they list?

**Comment by jakirkham** [comment]
Also, I'm still trying to understand the commit notation that is used here. Is `ENH` correct or should it be something else?

**Comment by charris** [comment]
I'd probably do `MAINT`, `ENH` is mostly new features, functions, keywords, etc., but the difference isn't worth worrying about.

**Comment by charris** [comment]
Merging, thanks @jakirkham .

**Comment by jakirkham** [comment]
Oh ok, I was going to change to `MAINT`. Sorry.

**Comment by jakirkham** [comment]
Thanks for your feedback @charris.

**Comment by charris** [comment]
You can add `[ci skip]` to the commit message if no tests are involved.

**Comment by charris** [comment]
PS, do you like PyCharm?

**Comment by jakirkham** [comment]
Ok, I'll keep that in mind.

Yeah, I'm pretty happy with it. It seems quite a bit faster and lighter than Eclipse. The quick open features are pretty nice. They have been working on iPython support in the recent version. It still a little rough around the edge, but it is a nice feature.

**Pull request merged**